server/main.js

```diff
@@ -157,36 +154,65 @@ function ensureSlackUserGroups({
       }
     ];
   }
-  return Promise.all(
-    onCalls.map(({ slackUserGroupHandle, onCallSlackUserID }) => {
-      if (!slackUserGroupHandle || !onCallSlackUserID) {
-        return;
-      }
-      const slackUserGroupID = slackUserGroupIDsByHandle.get(
-        slackUserGroupHandle
+
+  return promisify(slack.usergroups.list)({ token: slackToken }).then(
+    ({ usergroups }) => {
+      const userGroupInfoByHandle = new Map();
+      usergroups.forEach(({ id, handle, name }) =>
+        userGroupInfoByHandle.set(handle, { name, id })
       );
-      if (!slackUserGroupID) {
-        // Should be guaranteed at startup.
-        return Promise.reject(
-          Error(`Unknown user group handle ${slackUserGroupHandle}`)
-        );
-      }
-      return promisify(slack.usergroups.users.list)({
-        token: slackToken,
-        usergroup: slackUserGroupID
-      }).then(({ users }) => {
-        if (users.join(",") !== onCallSlackUserID) {
-          console.log(
-            `Updating usergroup ${slackUserGroupHandle} to ${onCallSlackUserID}`
-          );
-          return promisify(slack.usergroups.users.update)({
-            token: slackToken,
-            usergroup: slackUserGroupID,
-            users: onCallSlackUserID
-          });
-        }
-      });
-    })
+      return Promise.all(
+        onCalls.map(
+          ({ slackUserGroupHandle, onCallName, onCallSlackUserID }) => {
+            if (!(slackUserGroupHandle && onCallSlackUserID && onCallName)) {
+              return;
+            }
+            const userGroupInfo = userGroupInfoByHandle.get(
+              slackUserGroupHandle
+            );
+            if (!userGroupInfo) {
+              return Promise.reject(
+                Error(`Unknown user group handle ${slackUserGroupHandle}`)
+              );
+            }
+            const existingName = userGroupInfo.name || "";
+            const existingNamePrefix = existingName.replace(/\s*\(.*/, "");
```

Comment:
Should we catch the "" case and put in `"Oncall $level"` or something of the sort?

Reply:
done